packages/automated-dispute/src/providers/protocolProvider.ts

```diff
@@ -308,125 +291,289 @@ export class ProtocolProvider implements IProtocolProvider {
-     * @returns A promise that resolves to an array of EboEvents.
+     * @param {bigint} fromBlock - The starting block number to fetch events from.
+     * @param {bigint} toBlock - The ending block number to fetch events up to.
+     * @returns {Promise<EboEvent<"OracleRequestFinalized">[]>} A promise that resolves to an array of `OracleRequestFinalized` events.
+     * @throws {Error} If event block number or log index is null.
+     */
+    private async getOracleRequestFinalizedEvents(
+        fromBlock: bigint,
+        toBlock: bigint,
+    ): Promise<EboEvent<"OracleRequestFinalized">[]> {
+        const events = await this.l2ReadClient.getContractEvents({
+            address: this.oracleContract.address,
+            abi: oracleAbi,
+            eventName: "OracleRequestFinalized",
+            fromBlock,
+            toBlock,
+            strict: true,
+        });
+
+        return events.map((event) => {
+            if (event.blockNumber === null || event.logIndex === null) {
+                throw new Error("event.blockNumber or event.logIndex is null");
+            }
+
+            const { _requestId, _responseId, _caller } = event.args;
+
+            return {
+                name: "OracleRequestFinalized",
+                blockNumber: event.blockNumber,
+                logIndex: event.logIndex,
+                rawLog: event,
+
+                requestId: _requestId,
+                metadata: {
+                    requestId: _requestId,
+                    responseId: _responseId,
+                    caller: _caller,
+                },
+            } as unknown as EboEvent<"OracleRequestFinalized">;
+        });
+    }
+
+    /**
+     * Fetches all relevant events from the Oracle contract within a specified block range.
+     *
+     * @param {bigint} fromBlock - The starting block number to fetch events from.
+     * @param {bigint} toBlock - The ending block number to fetch events up to.
+     * @returns {Promise<EboEvent<EboEventName>[]>} A promise that resolves to an array of Oracle events.
+     */
+    async getOracleEvents(fromBlock: bigint, toBlock: bigint) {
+        const [
+            responseProposedEvents,
+            responseDisputedEvents,
+            disputeStatusUpdatedEvents,
+            disputeEscalatedEvents,
+            oracleRequestFinalizedEvents,
+        ] = await Promise.all([
+            this.getResponseProposedEvents(fromBlock, toBlock),
+            this.getResponseDisputedEvents(fromBlock, toBlock),
+            this.getDisputeStatusUpdatedEvents(fromBlock, toBlock),
+            this.getDisputeEscalatedEvents(fromBlock, toBlock),
+            this.getOracleRequestFinalizedEvents(fromBlock, toBlock),
+        ]);
+
+        return this.mergeEventStreams(
+            responseProposedEvents,
+            responseDisputedEvents,
+            disputeStatusUpdatedEvents,
+            disputeEscalatedEvents,
+            oracleRequestFinalizedEvents,
+        );
```

Comment:
We could probably end up returning a flattened array of events as they are going to be merged with EBORequestCreator events:
```suggestion
        return [
            ...responseProposedEvents,
            ...responseDisputedEvents,
            ...disputeStatusUpdatedEvents,
            ...disputeEscalatedEvents,
            ...oracleRequestFinalizedEvents,
        ];
```
This way we just sort the array once. I'd specify though in the method doc that no specific order is enforced in the returned array.